src/groupByKey.ts

```diff
@@ -2,6 +2,42 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const workItems: T[] = JSON.parse(JSON.stringify(items));
+  const groupOfObjects: GroupsMap<T> = {};
+
+  workItems.forEach((item) => {
+    const currentKey = item[key] as keyof GroupsMap<T>;
+
+    if (!Object.prototype.hasOwnProperty.call(groupOfObjects, currentKey)) {
+      groupOfObjects[currentKey] = [];
+    }
+
+    groupOfObjects[currentKey].push(item);
+  });
+
+  return groupOfObjects;
 }
+
+// type ForSet = string | number;
```

Comment:
definitely not the best name for this type =)
name should represent type, not the purpose of that type

but I agree that in that case it's hard to find what name will be better here

Reply:
I really try)) choose between KeyType and ForSet both not the best((